### PR TITLE
KEY-684: Export without secrets

### LINKED ISSUE #72: Export without secrets
Hello, 

Is it possible to export a yaml file stripping out the client secrets? It doesn't look like it. If not, it would be a great idea as we are looking to use an export of the current state before a deploy as an artifact for rollback and we wouldn't want it to contain sensitive info.

Thanks

### PR DESCRIPTION
**Pull request by fyockm**
## ✏️ Changes
Fixes #72 
- `--strip` option is replaced by `AUTH0_EXPORT_IDENTIFIERS: true`. It's now false by default, which means that IDs will be removed from exported object.
- `INCLUDED_PROPS` and `EXCLUDED_PROPS` are added to the config. Those can be used for flexible tweak of what properties should be included into exported objects.

## 🔗 References
  Jira: https://auth0team.atlassian.net/browse/KEY-684
  Github: #72
  Supersedes: #93 
  
## 🎯 Testing 
✅ This change has been tested locally
✅ This change has unit test coverage

### CODE REVIEW

File: src/args.js [hunk 1]
```diff
@@ -58,20 +58,20 @@ export default yargs
...
-      alias: 's',
-      describe: 'Strip the identifier field for each object type.',
-      type: 'boolean',
-      default: false
-    },
     secret: {
       alias: 'x',
       describe: 'The client secret, this allows you to encrypt the secret in your build configuration instead of storing it in a config file',
       type: 'string'
+    },
```
Comment (zxan1285) [inline src/args.js]: Probably should remove this. If we're going to use `AUTH0_EXPORT_IDENTIFIERS: true`

Comment (zxan1285) [reply src/args.js]: And we decided to keep `export_ids` param. If set, it'll override config value.

**Comment by zxan1285** [comment]
> Looks good to me except..
> 
> * I think we need an area which explains all the config options/cmd line options? The examples can ref to it? What do you think?
> * what about identifiers? I think we need to strip them by default? Similar to this, another PR??
> * I think we might want to being the config.json and cmd line params together, to me the cmdline params should just set the config, they should be the same?

1. https://auth0.com/docs/extensions/deploy-cli this seems like a nice place for all the explanations. Yet i'm not sure i'm the best person to explain anything :)

2. We do strip ids by default here: https://github.com/auth0/auth0-deploy-cli/pull/94/files#diff-f4f8238e49dd5edfd8bf15d981ed2802R109

3. Added that in the last commit. `--export_ids` will override config value.

**Comment by fyockm** [comment]
published as `auth0-deploy-cli@3.0.0`

**Pull request merged**